### PR TITLE
🌐 [i18n-KO] Translated `package_reference/serialization.md` to Korean

### PR DESCRIPTION
**Pull request by seoyoung-3060**
# What does this PR do?

Translated the `package_reference/serialization.md` file of the documentation to Korean.
Thank you in advance for your review. 

Part of https://github.com/huggingface/huggingface_hub/issues/1626

## Before reviewing
- [x] Check for missing / redundant translations (번역 누락/중복 검사)
- [x] Grammar Check (맞춤법 검사)
- [x] Review or Add new terms to glossary (용어 확인 및 추가)
- [x] Check Inline TOC (e.g. `[[lowercased-header]]`)
- [x] Check live-preview for gotchas (live-preview로 정상작동 확인)

## Who can review? (Initial)

Team Hugging Face KREW, may you please review this PR? 
@harheem, @cjfghk5697, @seoulsky-field, @nuatmochoi, @boyunJang, @SeungAhSon, @seoyoung-3060, @junejae, @usr-bin-ksh, @fabxoe

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review? (Final)

 @jungnerd, @heuristicwave May you please review this PR?

### CODE REVIEW

File: docs/source/ko/package_reference/serialization.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<!--⚠️ Note that this file is in Markdown but contain specific syntax for our doc-builder (similar to MDX) that may not be
+rendered properly in your Markdown viewer.
+-->
+
+# 직렬화[[serialization]]
+
+`huggingface_hub`에는 ML 라이브러리가 모델 가중치를 표준화된 방식으로 직렬화 할 수 있도록 돕는 헬퍼를 포함하고 있습니다. 이 라이브러리의 부분은 아직 개발 중이며 향후 버전에서 개선될 예정입니다. 이러한 개선의 목표는 Hub에서 가중치의 직렬화 방식을 통일하고, 라이브러리 간 코드 중복을 줄이며, Hub에서의 규약을 촉진하는 것입니다.
```
Comment (boyunJang) [inline docs/source/ko/package_reference/serialization.md:7]:
```suggestion
`huggingface_hub`에는 ML 라이브러리가 모델 가중치를 표준화된 방식으로 직렬화 할 수 있도록 돕는 헬퍼를 포함하고 있습니다. 라이브러리의 이 부분은 아직 개발 중이며 향후 버전에서 개선될 예정입니다. 개선 목표는 Hub에서 가중치의 직렬화 방식을 통일하고, 라이브러리 간 코드 중복을 줄이며, Hub에서의 규약을 촉진하는 것입니다.
```
문장이 어색한 부분만 조금씩 수정했습니다.

File: docs/source/ko/package_reference/serialization.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+## 상태 사전을 조각으로 나누기[[split-state-dict-into-shards]]
```
Comment (boyunJang) [inline docs/source/ko/package_reference/serialization.md:9]:
```suggestion
## 상태 사전을 샤드로 나누기[[split-state-dict-into-shards]]
```
shard는 "샤드"로 두는 게 나을 것 같아서 수정했습니다!

File: docs/source/ko/package_reference/serialization.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+현재 이 모듈은 상태 딕셔너리(예: 레이어 이름과 관련 텐서 간의 매핑)를 받아 여러 조각으로 나누고, 이 과정에서 적절한 인덱스를 생성하는 단일 헬퍼를 포함하고 있습니다. 이 헬퍼는 torch, tensorflow, numpy 텐서에 사용 가능하며, 다른 ML 프레임워크로 쉽게 확장될 수 있도록 설계되었습니다.
```
Comment (boyunJang) [inline docs/source/ko/package_reference/serialization.md:11]:
```suggestion
현재 이 모듈은 상태 딕셔너리(예: 레이어 이름과 관련 텐서 간의 매핑)를 받아 여러 샤드로 나누고, 이 과정에서 적절한 인덱스를 생성하는 단일 헬퍼를 포함하고 있습니다. 이 헬퍼는 `torch`, `tensorflow`, `numpy` 텐서에 사용 가능하며, 다른 ML 프레임워크로 쉽게 확장될 수 있도록 설계되었습니다.
```
조각 -> 샤드로 수정했습니다.
역슬래시(`)가 빠져있길래 추가했습니다.

File: docs/source/ko/package_reference/serialization.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+[[autodoc]] huggingface_hub.split_tf_state_dict_into_shards
+
+### split_torch_state_dict_into_shards[[huggingface_hub.split_torch_state_dict_into_shards]]
+
+[[autodoc]] huggingface_hub.split_torch_state_dict_into_shards
+
+### split_state_dict_into_shards_factory[[huggingface_hub.split_state_dict_into_shards_factory]]
+
+이것은 각 프레임워크별 헬퍼가 파생되는 기본 틀입니다. 실제로는 아직 지원되지 않는 프레임워크에 맞게 조정할 필요가 있는 경우가 아니면 이 틀을 직접 사용할 것으로 예상되지 않습니다. 그런 경우가 있다면, `huggingface_hub` 리포지토리에 [새로운 이슈를 개설](https://github.com/huggingface/huggingface_hub/issues/new) 하여 알려주세요.
```
Comment (boyunJang) [inline docs/source/ko/package_reference/serialization.md:27]:
```suggestion
이것은 각 프레임워크별 헬퍼가 파생되는 기본 틀입니다. 아직 지원되지 않는 프레임워크에 맞게 조정할 필요가 있는 경우가 아니면 이 틀을 직접 사용할 것으로 예상되지 않습니다. 그런 경우가 있다면, `huggingface_hub` 저장소에 [새로운 이슈를 개설](https://github.com/huggingface/huggingface_hub/issues/new) 하여 알려주세요.
```
국문 조금 수정했습니다.
glossary에 repository는 저장소로 번역하길래 수정해뒀습니다.

**Review by boyunJang — COMMENTED** [review]
번역 작업하느라 정말 수고하셨습니다! 단어들만 조금 수정했습니다:)

File: docs/source/ko/package_reference/serialization.md (hunk 4 above)
Comment (seoulsky-field) [reply docs/source/ko/package_reference/serialization.md:27]: 참고하는 glossary마다 조금씩 다른 것 같습니다! 이번에는 대부분 리포지토리로 번역했던 것 같아서 논의해보면 좋을 것 같습니다!

Comment (seoyoung-3060) [reply docs/source/ko/package_reference/serialization.md:27]: 좋은 의견공유 감사합니다. 
맞아요 glossary가 좀 섞여있더라구요.. 
이번에는 모두 리포지토리로 번역이 진행되었어서, 리포지토리로 번역하는게 나을 것 같다고 생각됩니다!

**Comment by heuristicwave** [comment]
[@Wauplin](https://github.com/Wauplin) I've reviewed this PR, and it looks good to me.  
Would you mind merging this PR, please?

**Review by Wauplin — APPROVED** [review]
Thanks! I'll merge it right away :)

**Pull request merged**